beacon_node/network/src/sync/manager.rs

```diff
@@ -107,6 +107,9 @@ pub enum SyncMessage<E: EthSpec> {
     /// A blob with an unknown parent has been received.
     UnknownParentBlob(PeerId, Arc<BlobSidecar<E>>),
 
+    /// A data column with an unknown parent has been received.
+    UnknownParentDataColumn(PeerId, Arc<DataColumnSidecar<E>>),
```

Comment:
I think you're referring to `GossipDataColumnError::ParentUnknown` instead of `SyncMessage` here? we do need to keep the data column in the `BlockComponent` when doing a parent lookup.

Reply:
I've removed the unused error types in `GossipDataColumnError` and leaving this line unchanged. Let me know if you still want the payload removed for this error?